### PR TITLE
[API] on 500 error only show message if gitea in dev mode

### PR DESCRIPTION
**Pull request by 6543**
depend on https://github.com/go-gitea/gitea/pull/11643

### CODE REVIEW

**Comment by 6543** [comment]
**CI** will fail as long as there exist **expected** errors witch return 500 httpStatus!!!

so Update this branch until the point is where CI pass :)
StatusCode changes should happen in other pulls

**Comment by 6543** [comment]
next pull: #11644

File: modules/context/api.go [hunk 1]
```diff
@@ -69,6 +69,7 @@ type APIRedirect struct{}
 // If status is 500, also it prints error to log.
 func (ctx *APIContext) Error(status int, title string, obj interface{}) {
 	var message string
+
```
Comment (6543) [inline modules/context/api.go]:
```suggestion
```

File: modules/context/api.go [hunk 2]
```diff
@@ -77,6 +77,10 @@ func (ctx *APIContext) Error(status int, title string, obj interface{}) {
 
 	if status == http.StatusInternalServerError {
 		log.ErrorWithSkip(1, "%s: %s", title, message)
+
+		if macaron.Env == macaron.PROD {
+			message = ""
```
Comment (6543) [inline modules/context/api.go:82]: we may should leave a hint to enable DEV for debugging?

**Comment by 6543** [comment]
it should pass if #11643 was merged

File: modules/context/api.go (hunk 2 above)
Comment (zeripath) [reply modules/context/api.go:82]: Meh. I mean it'll be the same as the other ISEs

**Review by guillep2k — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Comment by 6543** [comment]
ready to fly 🚀

**Comment by techknowlogick** [comment]
@6543 please backport :)

**Comment by 6543** [comment]
Feel free to do so, cant di it now

**Comment by 6543** [comment]
@techknowlogick had time to create it myselfe #11753

**Pull request merged**